lib/billy/proxy_connection.rb

```diff
@@ -74,12 +74,25 @@ def handle_request
 
     private
 
+    def prepare_response_headers_for_evma_httpserver(headers)
+      # Remove the headers below because they will be added later by evma_httpserver (EventMachine::DelegatedHttpResponse).
+      # See https://github.com/eventmachine/evma_httpserver/blob/master/lib/evma_httpserver/response.rb
+      headers_to_remove = [
+        'transfer-encoding',
+        'content-length',
+        'content-encoding'
```

Comment:
If we don't manually remove it, the response still includes a `Content-Encoding: gzip` header, which isn't valid since Puffing Billy forces it to binary (see this line https://github.com/oesmith/puffing-billy/blob/master/lib/billy/handlers/proxy_handler.rb#L76). 

Anyway, the most convincing argument is that if we don't remove it, the browser will throw a `Content Encoding Error` ([Firefox screenshot](http://imgur.com/IZzEX6t)), and if we do remove it, everything works correctly.